cmd/podman/pod_create.go

```diff
@@ -102,6 +107,10 @@ func podCreateCmd(c *cli.Context) error {
 		defer podIdFile.Close()
 		defer podIdFile.Sync()
 	}
+
+	if !c.BoolT("infra") && len(c.StringSlice("publish")) > 0 {
+		return errors.Errorf("you must have an infra container to publish port bindings to the host")
+	}
```

Comment:
could you also add an error when we are in rootless mode?  Exposing port from the host is not currently supported for rootless containers.

Reply:
sure! added